### ISSUE TITLE
Impossible to submit file when updating dataset

### ISSUE THREAD

**Issue by lilaspourpre** [codalab/codalab-competitions#3094]
Dear Support,

We are organising a competition https://codalab.lisn.upsaclay.fr/competitions/642 and we updated our test set for the next phase. We cannot check that everything will work correctly, so I created a new competition to check this. However, this is the error that I get:

![image](https://user-images.githubusercontent.com/9992898/151538651-764089ad-4550-49ee-9a8e-da041f6039b1.png)

The error is not about the file that I am submitting. I am afraid that this error will also occur with the 642 competition on Monday.

Yours sincerely,
Irina Nikishina.

**Comment by Didayolo** [comment]
Hello,

To clarify, when you have creating the new test competition, you upload a clone of your competition? And then you have updated the test dataset on this clone?

Or did you upload the new competition with the new test set?

**Comment by Didayolo** [comment]
According to this past issue:
- #2353

It seems that this error, even if the error message is unclear, shows up when the file is corrupted. Did you double check everything was right with the new dataset?

To let me help you, maybe you can:
- Add me as an admin of your test competition (my CodaLab ID is "pavao")
- Send me the ZIP file in question (I know you can't share it publicly, but if you'd like to you can share it privately)

**Comment by lilaspourpre** [comment]
I have already added you to admins (as it has been requested when opening the issue). And yes, I would like to send it to you privately. Not sure, if it is possible on Github

**Comment by Didayolo** [comment]
Here is my email address: adrien.pavao@gmail.com

**Comment by Didayolo** [comment]
I can see that you didn't do any other submission on the new competition. Can you please check that the other phases/datasets are working as intended?

**Comment by lilaspourpre** [comment]
Dear @Didayolo , 

We updated the dataset in the real competition https://codalab.lisn.upsaclay.fr/competitions/642
and got the following error:

![image](https://user-images.githubusercontent.com/9992898/151749878-69177cd0-1740-4813-bb5e-abf33baa91e3.png)

The error says that there are special characters in the file, however, we deleted them and resubmitted the dataset.

I sent you files via email (both reference and submission). I also made you an admin of the competition.

Thank you in advance for your help.
Yours sincerely, 

Irina Nikishina.

**Comment by lilaspourpre** [comment]
Dear @Didayolo ,

We created another working copy of competition #642 (https://codalab.lisn.upsaclay.fr/competitions/1764). We use the same reference data for the final test phase, however, it works in #1764 but not in #642.

**Comment by lilaspourpre** [comment]
@Tthomas63, maybe you could assist us with the problem?

**Comment by lilaspourpre** [comment]
We assume that the test file is somehow cashed in the competition, could you please help us with that?

**Comment by lilaspourpre** [comment]
@Didayolo @Tthomas63

**Comment by Didayolo** [comment]
Hi,

Thank you for sending the files in the email.

> We use the same reference data for the final test phase, however, it works in #1764 but not in #642.

It is indeed strange if it works in one case and not the other. What could be the difference?

- Was the test phase already existent in the challenge 642 or did you create it using the editor?
- Are you sure the scoring programs are exactly similar? (you have `scoring_program_russe_2022_verify` and `scoring_program_1_1_1764`)
- It is normal that "phase never ends" is checked in the configuration of the competition 642?

**Comment by Didayolo** [comment]
Also,

> We updated the dataset in the real competition and got the following error [...]

What is the ID of the submission in question? Was it done on the development phase or the test phase?

**Comment by lilaspourpre** [comment]
`Was the test phase already existent in the challenge 642 or did you create it using the editor?`
Yes, it was created since the competition was created using .yaml

`Are you sure the scoring programs are exactly similar? (you have scoring_program_russe_2022_verify and scoring_program_1_1_1764)`

Yes, it is a copy (dump) of our scoring program in the competition #642

`It is normal that "phase never ends" is checked in the configuration of the competition 642?`
Yes, we want to leave it open

**Comment by lilaspourpre** [comment]
`What is the ID of the submission in question? Was it done on the development phase or the test phase?`

test data, I do not know how to look at the submission ID

![image](https://user-images.githubusercontent.com/9992898/151975643-7223954a-78d9-4eeb-9abf-f2d5c80c0da9.png)

**Comment by Didayolo** [comment]
> Yes, it is a copy (dump) of our scoring program in the competition #642

Indeed, I just replaced the scoring program (in the test competition 1764 of course) by `scoring_program_russe_2022_verify` and the submission on test phase worked.

> test data, I do not know how to look at the submission ID

Test data but replaced on the development phase?

You have an admin panel "Submissions" in which you can track all submissions made by participants (including yours).

**Comment by lilaspourpre** [comment]
the development phase?

no, the test phase

**Comment by lilaspourpre** [comment]
42472 is the submission id

**Comment by Didayolo** [comment]
OK now I see it but it's a new submission isn't it?

**Comment by Didayolo** [comment]
The strange part is that the error is not in the "scoring error log"

**Comment by lilaspourpre** [comment]
I have just recreated the error for you to see it

**Comment by Didayolo** [comment]
I think I've solved the issue!

For some reason, "result scoring only" needed to be checked in the test phase for it to work.

It is strange since it is not supposed to be necessary. Submissions without a "metadata" file included are interpreted (or are supposed to) as results submissions (as opposed to code submissions).

Maybe the fact that "results scoring only" was checked on the development phase caused the issue. In your competition clone, the setting was not checked at all.

**Comment by lilaspourpre** [comment]
Does it work now in #642? Let me check

**Comment by lilaspourpre** [comment]
@Didayolo it worked! But I do not get how "result scoring only" is connected to the evaluation and this error? Could you please elaborate on that?

**Comment by lilaspourpre** [comment]
However, I cannot see the result on the leaderboard

**Comment by lilaspourpre** [comment]
There is no button and it is not automatically displayed

**Comment by Didayolo** [comment]
> Could you please elaborate on that?

My hypothesis is that the submissions were interpreted as code submissions. I could tell that because the error showed on the file `[...]/res/input/metadata` which does not exist (there is no `metadata` file in your submission). This file is included only on code submissions.

I think that, for some reason, the "result scoring only" option messed up.

> However, I cannot see the result on the leaderboard
> There is no button and it is not automatically displayed

Please have a look at the different options in the editor, such as "force submission to leaderboard", "allow leaderboard modifying", etc. Tell me then if there is something specific that does not work.

**Comment by Didayolo** [comment]
@lilaspourpre any update on this?

**Comment by Didayolo** [comment]
I close this issue as it is staling and specific to one user. Feel free to re-open it if needed.
